admin-dev/themes/new-theme/js/pages/order/view/order-product-add.js

```diff
@@ -180,4 +182,28 @@ export default class OrderProductAdd {
       }
     });
   }
+
+  handleAddProductWithConfirmationModal(event) {
+    const invoiceId = parseInt(this.invoiceSelect.val(), 10);
+    const orderId = $(event.currentTarget).data('orderId');
+
+    if (invoiceId === 0) {
+      const modalTranslations = this.invoiceSelect.data('modal-translations');
+      const modal = new ConfirmModal({
+        id: 'modal-confirm-new-invoice',
+        confirmTitle: modalTranslations['modal.title'],
+        confirmMessage: modalTranslations['modal.body'],
+        confirmButtonLabel: modalTranslations['modal.apply'],
+        closeButtonLabel: modalTranslations['modal.cancel'],
+      }, () => this.onAddProductWithNewInvoiceClick(orderId));
+
+      modal.show();
+    } else {
+      this.addProduct(orderId);
+    }
+  }
+
+  async onAddProductWithNewInvoiceClick(orderId) {
```

Comment:
Out of curiosity, why do you need async/await here, but not on `handleAddProductWithConfirmationModal` when you call the same `addProduct` method?

Reply:
Indeed its useless. Modified :heavy_check_mark: